book/scheduling.md

```diff
@@ -698,6 +690,9 @@ class JSContext:
 ```
 
 There are more calls to `render`; you should find and fix all of them.
+However, there is one call to `render` at the beginning of `click` that
+should remain, because it's necessary to make sure that layout is up-to-date
+before finding which layout object is under the click.
```

Comment:
I think this still isn't right. I went back through and Ch 9 is the last one that touches `click` (it adds JS event dispatching). But Ch9 doesn't render at the top of `click`—because in Chapter 9 we eagerly render any time we change, so there's no need to do it at the top of `click`. I think this paragraph should be something like:

> Now we don't immediately render on change. That means that the layout tree (and style) could be out of date when a method is called. Normally, this isn't a problem, but in one important case it is: click handling. That's because we need to *read* the layout tree to figure out what object was clicked on, which means the layout tree needs to be up to date. Add a call to `render` at the top of `click`:
> ... code block here